src/FormElement/ScheduleElement/MonthlyFields.php

```diff
@@ -0,0 +1,175 @@
+     */
+    public function setDefault(int $default): self
+    {
+        $this->default = $default;
+
+        return $this;
+    }
+
+    /**
+     * Get all the selected weekdays
+     *
+     * @return array
+     */
+    public function getSelectedDays(): array
+    {
+        $selectedDays = [];
+        foreach (range(1, $this->availableFields) as $day) {
+            if ($this->getValue("day$day", 'n') === 'y') {
+                $selectedDays[] = $day;
+            }
+        }
+
+        if (empty($selectedDays)) {
+            $selectedDays[] = $this->default;
+        }
+
+        return $selectedDays;
+    }
+
+    protected function assemble()
+    {
+        $this->getAttributes()->set('id', $this->protectId('monthly-fields'));
+
+        $runsOn = $this->getPopulatedValue('runsOn', static::RUNS_EACH);
+        $this->addElement('radio', 'runsOn', [
+            'required' => true,
+            'class'    => 'autosubmit',
+            'value'    => $runsOn,
+            'options'  => [static::RUNS_EACH => $this->translate('Each')],
+        ]);
+
+        $listItems = HtmlElement::create('ul', ['class' => ['schedule-element-fields', 'multiple-fields']]);
+        if ($runsOn === static::RUNS_ONTHE) {
+            $listItems->getAttributes()->add('class', 'disabled');
+        }
+
+        $foundCheckedDay = false;
+        foreach (range(1, $this->availableFields) as $day) {
+            $checkbox = $this->createElement('checkbox', "day$day", [
+                'class' => 'sr-only autosubmit',
+                'value' => $this->getPopulatedValue("day$day", 'n')
+            ]);
+            $this->registerElement($checkbox);
+
+            $foundCheckedDay = $foundCheckedDay || $checkbox->isChecked();
+            $htmlId = $this->protectId("day$day");
+            $checkbox->getAttributes()->set('id', $htmlId);
+
+            $listItem = HtmlElement::create('li');
+            $checkbox->prependWrapper($listItem);
+
+            $listItem->addHtml($checkbox, HtmlElement::create('label', ['for' => $htmlId], $day));
+            $listItems->addHtml($checkbox);
+        }
+
+        if (! $foundCheckedDay) {
+            $this->getElement("day{$this->default}")->setChecked(true);
+        }
+
+        $monthlyWrapper = HtmlElement::create('div', ['class' => 'monthly']);
+        $runsEach = $this->getElement('runsOn');
+        $runsEach->prependWrapper($monthlyWrapper);
+        $monthlyWrapper->addHtml($runsEach, $listItems);
+
+        $this->addElement('radio', 'runsOn', [
+            'required' => $runsOn !== static::RUNS_EACH,
+            'class'    => 'autosubmit',
+            'options'  => [static::RUNS_ONTHE => $this->translate('On the')]
+        ]);
+
+        $runsOnThe = $this->getElement('runsOn');
+        $runsOnValidators = $runsOnThe->getValidators();
+        $runsOnValidators
+            ->clearValidators()
+            ->add(
+                new DeferredInArrayValidator(function (): array {
```

Comment:
Because I want to postpone the validation for later.

![Bildschirm­foto 2023-02-03 um 14 32 21](https://user-images.githubusercontent.com/57616252/216615946-f80787c2-e248-4109-80ae-8c6ac55414e1.png)

Reply:
Never mind! I have to set the `haystack` key explicitly.